File: 0901-1000/0978-Longest Turbulent Subarray/0978-Longest Turbulent Subarray.cpp
class Solution {
public:
    int maxTurbulenceSize(vector<int>& A) {
        int maxLen = 1, inc = 1, dec = 1;
        for (int i = 1; i < A.size(); ++i) {
            if (A[i] > A[i - 1]) {
                inc = dec + 1;
                maxLen = max(maxLen, inc);
                dec = 1;
            }
            else if (A[i] < A[i - 1]) {
                dec = inc + 1;
                maxLen = max(maxLen, dec);
                inc = 1;
            }
            else {
                inc = dec = 1;
            }
        }
        return maxLen;
    }
};

File: 0901-1000/0978-Longest Turbulent Subarray/0978-Longest Turbulent Subarray.py
class Solution:
    def maxTurbulenceSize(self, A: List[int]) -> int:
        maxLen = inc = dec = 1
        for i in range(1, len(A)):
            if A[i] > A[i - 1]:
                inc = dec + 1
                maxLen = max(maxLen, inc)
                dec = 1
            elif A[i] < A[i - 1]:
                dec = inc + 1
                maxLen = max(maxLen, dec)
                inc = 1
            else:
                inc = dec = 1
        return maxLen
